workers/unity/Assets/Gdk/Core/Systems/CleanReactiveComponentsSystem.cs

```diff
@@ -35,37 +38,64 @@ private void GenerateComponentGroups()
                     translationUnit.CleanUpComponentGroups.Add(GetComponentGroup(componentType));
                 }
             }
+
+            MethodInfo addRemoveComponentActionMethod =
+                typeof(CleanReactiveComponentsSystem).GetMethod("AddRemoveComponentAction",
+                    BindingFlags.NonPublic | BindingFlags.Instance);
+
+            // Find all components with the RemoveAtEndOfTick attribute
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                foreach (var type in assembly.GetTypes())
+                {
+                    if (type.GetCustomAttribute<RemoveAtEndOfTick>(false) == null
+                        || typesToRemove.Contains(type)
```

Comment:
I recommend moving `typesToRemove.Contains` check at the end of this conditional block, or even after it as a separate `continue`. This is because it's currently between two type-based conditions

Reply:
Will do